### PR TITLE
Revamped snackbar icons, migrated to common.js, made freeze objects f…

### LINKED ISSUE #135: Migrate snackbar functions to common.js and optimizing its functions.

### PR DESCRIPTION
**Pull request by Mohammed-JS-Khan**
…or icons. Fixes #135

### CODE REVIEW

File: public/js/common.js [hunk 1]
```diff
@@ -11,6 +11,13 @@ const colours = Object.freeze({
     redLight      : 'rgba(243, 13, 20, 0.5)'
 });
 
+const snack = Object.freeze({
+    success     :   '<i class="material-icons">check</i>' + '&nbsp&nbsp&nbsp',
+    warning     :   '<i class="material-icons">warning</i>' + '&nbsp&nbsp&nbsp',
+    fail        :   '<i class="material-icons">block</i>' + '&nbsp&nbsp&nbsp',
+    close       :   '&nbsp&nbsp&nbsp' + '<i id=closeSnack class="material-icons">close</i>'
```
Comment (ahmedqarmout2) [inline public/js/common.js]: why the plus? why not just have each as one string instead of addition of 2 strings?

**Review by ahmedqarmout2 — APPROVED** [review]
good stuff

File: public/js/common.js (hunk 1 above)
Comment (Mohammed-JS-Khan) [reply public/js/common.js]: cause im silly :)

Comment (Mohammed-JS-Khan) [reply public/js/common.js]: fixed.

File: public/js/admin.js [hunk 2]
```diff
@@ -274,16 +274,13 @@ var deactivateUser = function(id) {
...
-                dropSnack(colour, icon + id + ' account' + msg);
+                warningSnackbar(id + ' account' + msg);
             },
             error: function(data){
                 if (data['status'] === 401) {
                     window.location.href = '/';
                 } else {
-                    const icon = '<i class="material-icons">cancel</i>&nbsp&nbsp&nbsp';
-                    dropSnack(colours.redDark, icon + 'Failed to deactivate ' + id + '\' account');
+                    failSnackbar('Failed to deactivate ' + id + '\' account');
```
Comment (ahmedqarmout2) [inline public/js/admin.js]: Id \'s account

**Review by ahmedqarmout2 — APPROVED** [review]

File: public/js/admin.js [hunk 3]
```diff
@@ -311,16 +308,13 @@ var activateUser = function(id) {
             success: function(data) {
                 displayAccountsTable();
                 const msg = ' has been activated';
-                const colour = colours.green;
-                const icon = '<i class="material-icons">check</i>&nbsp&nbsp&nbsp';
-                dropSnack(colour, icon + id + ' account' + msg);
+                successSnackbar(id + ' account' + msg);
```
Comment (esteeroramy) [inline public/js/admin.js]: Id 's account

File: public/js/admin.js [hunk 4]
```diff
@@ -435,15 +429,15 @@ var submitEditForm = function(id) {
                 submitEditForm(user.id ? user.id : id);
             });
             displayAccountsTable();
-            dropSnack(colours.green, 'User ' + id + ' has been updated');
+            successSnackbar('User ' + id + ' has been updated');
```
Comment (esteeroramy) [inline public/js/admin.js]: Id 's account

File: public/js/admin.js (hunk 3 above)
Comment (esteeroramy) [reply public/js/admin.js]: why do we need variable `msg` in this case? why cant we just have `successSnackbar(id + '\'s account has been activated');`

**Review by esteeroramy — CHANGES_REQUESTED** [review]
Minor comment, besides that this looks good, will approve as soon as addressed, good work.

File: public/js/admin.js (hunk 3 above)
Comment (Mohammed-JS-Khan) [reply public/js/admin.js]: Not sure, that was there from before. I thought someone wanted to add some extra functionality there later. Ill mend it. 👍

**Review by esteeroramy — APPROVED** [review]

**Review by mehtadarshan1 — APPROVED** [review]
LOOKS GOOD

**Pull request merged**